### PR TITLE
[#817] Add `:slow` tags to exclude tests

### PR DESCRIPTION
**Pull request by jiegillet**
Solves #817.

The "slow" test for `prime-factors` is really silly, there are only two factors and the first one is 5. Of course 5 will get found quickly, even by looking at the number you can tell. A real test would have two large factors. Anyway.

Still WIP, because I would like to add a `instructions.append.md` file to each of these and I want to get the wording right before I add the files in case I have to change it. How about this:

> One or several of the tests of this exercise have been tagged as `:slow`, which means that they will get skipped by the automated test runner, since some solutions may take cause a time out. Download the exercise locally to run all the tests on your machine if you want to test the efficiency of your solution.

Should we add that some optimized solutions exist for some specific ones? `pythagorean-triplet` already has a hint, and for `prime-factors`, the more naive the implementation the more likely it is to pass. The others take a while to pass for the example implementations. Maybe `list-ops` could use a mention of `foldr` to avoid using `reverse` (although I have not benchmarked that).

### CODE REVIEW

**Comment by angelikatyborska** [comment]
> Still WIP, because I would like to add a instructions.append.md file to each of these and I want to get the wording right before I add the files in case I have to change it.

Great idea!

About the content of that comment: it sounds good but it needs to make sense both for students that read it on the platform and that already downloaded the exercise. I would also rephrase it to first say why those tests are "slow" and only then how that affects the student. Here's my proposal:

> One or several of the tests of this exercise have been tagged as `:slow`. Those tests might take a long time to finish if your solution is not optimal. For this reason, they will not be run on the platform by the automated test runner. If you are solving this exercise directly on the platform in the web editor, you might want to consider downloading this exercise to your machine instead. This will allow you to run all the tests and check the efficiency of your solution.

> Should we add that some optimized solutions exist for some specific ones?

Only if you are sure of what to add there and that it would be correct. I wouldn't know 🤷 

>  Maybe list-ops could use a mention of foldr to avoid using reverse (although I have not benchmarked that).

`foldr` is one of the functions that the student must implement now after the exercise got updated to match problem specs. No students attempted that yet because the updated exercise wasn't deployed. For this reason, I would hold off from adding any hints to this one for now.

**Comment by jiegillet** [comment]
> Those tests might take a long time to finish if your solution is not optimal.

I stayed away from that wording because it implies that there is an optimal solution. There is one for some, like  `pythagorean-triplet `, but we don't have a solution for `palindrome-products` that finishes in 10 seconds. 

How about

> Those tests involve large input values and may take a long time to finish especially for non-optimal solutions.

**Comment by angelikatyborska** [comment]
> I stayed away from that wording because it implies that there is an optimal solution.

right 🤔 then let's just write:

> Those tests might take a long time to finish.

Without mentioning anything about the possibility of making them pass fast or not. It's probably something they would need to ask the mentor about anyway.

I don't like the phrase "large inputs" because it might suggest that the slowness comes from the input itself taking up a lot of space in memory, which is not true for most of those exercises.

**Review by angelikatyborska — APPROVED** [review]
Great! Thank you

**Comment by angelikatyborska** [comment]
(I haven't thought this through yet, just I'm writing down my observation)

We will need to fix the test runner because it reports skipped tests as failed.

I also think the append files could use a heading, but I'm not sure ATM how heading levels work with appends files. I will need to ask or read the website source.

Screen Shot 2021-07-30 at 19 01 29

Screen Shot 2021-07-30 at 19 01 44

**Comment by angelikatyborska** [comment]
I opened PRs for both problems
